icu4j/main/classes/core/src/com/ibm/icu/impl/units/ConversionRates.java

```diff
@@ -0,0 +1,243 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+/*
+ *******************************************************************************
+ * Copyright (C) 2004-2020, Google Inc, International Business Machines
+ * Corporation and others. All Rights Reserved.
+ *******************************************************************************
+ */
+
+package com.ibm.icu.impl.units;
+
+import com.ibm.icu.impl.Assert;
+import com.ibm.icu.impl.ICUData;
+import com.ibm.icu.impl.ICUResourceBundle;
+import com.ibm.icu.impl.UResource;
+import com.ibm.icu.util.MeasureUnit;
+import com.ibm.icu.util.UResourceBundle;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.TreeMap;
+
+public class ConversionRates {
+
+    public ConversionRates() {
+        // Read the conversion rates from the data (units.txt).
+        ICUResourceBundle resource;
+        resource = (ICUResourceBundle) UResourceBundle.getBundleInstance(ICUData.ICU_BASE_NAME, "units");
+        ConversionRatesSink sink = new ConversionRatesSink();
+        resource.getAllItemsWithFallback(UnitsData.Constants.CONVERSION_UNIT_TABLE_NAME, sink);
+        this.mapToConversionRate = sink.getMapToConversionRate();
+    }
+
+    /**
+     * Extracts the factor from a `SingleUnitImpl` to its Basic Unit.
+     *
+     * @param singleUnit
+     * @return
+     */
+    private UnitConverter.Factor getFactorToBase(SingleUnitImpl singleUnit) {
```

Comment:
I'm stricter than Shane: I think it makes sense, I have more time for Units.

If we want to make C++ look like this code in the future, then I need this code to be at least as good as, or better than, the C++ code. This also makes me additionally nitpicky on the naming. (One example: "Basic Unit" is not a concept on Wikipedia. "Base Unit" is.)

If getting this clean now is too much work for the release deadline, and we rather want to rush it out the door with technical debt to be fixed next quarter (which is more work than getting it done now, but does help us meet the deadline), then update the ticket to not say "make C++ look like Java" - it would be about revisiting and cleaning up both C++ and Java.

Reply:
@sffc Sorry that I took so long to review this PR thoroughly!
@younies Please don't hesitate to tell me "please review my PR as soon as you get the chance". Don't worry about creating pressure for me: for me, I think I do better with that pressure.

(Marking as "unresolved" because I continued this conversation.)

Reply:
(I also wasn't sure if it's best to get all code review feedback in one go, or whether it's better to iteratively go for a few rounds. I'm beginning to believe "all in one go" is best if/when the reviewer has sufficient capacity, while incremental makes sense if initial feedback is more urgently desired.)